### PR TITLE
Ports Laying down from Cosmatic Drift.

### PR DESCRIPTION
**Pull request by Adrian16199**
## About the PR

Ports laying down.
## Why / Balance

Because its the greatest thing known to man. (And woman)
## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

https://github.com/DeltaV-Station/Delta-v/assets/144424013/e059d531-f199-4612-8a94-10aade38b4c5

## Breaking changes

**Changelog**

:cl:
- add: You can now lay down and stand up.

### CODE REVIEW

File: Content.Server/_CD/Traits/LayEmoteComponent.cs
Comment (deltanedas) [inline Content.Server/_CD/Traits/LayEmoteComponent.cs:13]: [DataField] and use ProtoId

File: Content.Server/_CD/Traits/LayEmoteComponent.cs
Comment (deltanedas) [inline Content.Server/_CD/Traits/LayEmoteComponent.cs:19]: dont need = false

File: Content.Server/_CD/Traits/LayEmoteComponent.cs
Comment (deltanedas) [inline Content.Server/_CD/Traits/LayEmoteComponent.cs:16]: above

**Comment by Mnemotechnician** [comment]
But... why an emote and not a keybind, or button, or anything else that would make more sense for an action that affects gameplay?

**Comment by Timemaster99** [comment]
I agree with the above comment. This should be an action alongside sleeping, screaming, and harm mode

**Comment by Adrian16199** [comment]
"Affects gameplay"
Im just gonna ask, how exactly compared to:
Harm mode: Decides whether you are playing TDM (team deatch match), RDM if you are feeling funny or roleplayin game.
Scream: Somethin to indicate you are in distress that isnt neccesarily you spamming emotes or copy pastin scream line while someone is beating you upside the head. Also the sound itself can be heard better than anything else.
Sleeping: Used to cure diseases most of the time and revenants are capable of eating people who are sleeping like a sleeping demon, can only be done in bed.
Closest thing that I can think of affectin gameplay is the fact that you lock yourself in place... and then maybe pretend that you seized up, falled limb, your eyes, dead and lifeless.

Now code wise an action for a mere emote that can simply be keybinded like any other emote would require:
-A sprite for said action
-Coding the action itself
-Coding the fact that it would need to toggle between laying down and THEN standing up. 

I cant be bothered for somethin that im simply portin, wantin it ingame as fast as possible because I adore this emote by itself and for somethin that I cant agree with wholeheartedly when it can be done with a simple keybind that already exists for emotes.

**Comment by DangerRevolution** [comment]
> "Affects gameplay" Im just gonna ask, how exactly compared to: Harm mode: Decides whether you are playing TDM (team deatch match), RDM if you are feeling funny or roleplayin game. Scream: Somethin to indicate you are in distress that isnt neccesarily you spamming emotes or copy pastin scream line while someone is beating you upside the head. Also the sound itself can be heard better than anything else. Sleeping: Used to cure diseases most of the time and revenants are capable of eating people who are sleeping like a sleeping demon, can only be done in bed.

Not sure why we're filled with hatred towards actions but this:

> Closest thing that I can think of affectin gameplay is the fact that you lock yourself in place... and then maybe pretend that you seized up, falled limb, your eyes, dead and lifeless.

is the actual reason why this shouldn't be an action, this is a half-cooked feature that shouldn't be an action and should be relegated to being an HRP emote

File: Content.Server/_CD/Traits/LayEmoteSystem.cs
Comment (DangerRevolution) [inline Content.Server/_CD/Traits/LayEmoteSystem.cs:75]: worth considering to give them a tiny amount of movement speed for HRP crawling

File: Resources/Prototypes/_CD/Voice/lay_emotes.yml
Comment (DangerRevolution) [inline Resources/Prototypes/_CD/Voice/lay_emotes.yml:5]: this weirdly doesn't allow any other emote *except* voluntarily lying down... not sure what the reasoning was on CD but what I want to fall over? be pushed over? :O

Comment (Adrian16199) [reply Resources/Prototypes/_CD/Voice/lay_emotes.yml:5]: Good point, I could add more chat triggers like fell, got pushed over, etc etc.

File: Content.Server/_CD/Traits/LayEmoteSystem.cs
Comment (Adrian16199) [reply Content.Server/_CD/Traits/LayEmoteSystem.cs:75]: Hmmmmmmmmmmmmmm...... maybe...... maybe...
That is worth considering.

File: Resources/Prototypes/_CD/Voice/lay_emotes.yml
Comment (DangerRevolution) [inline Resources/Prototypes/_CD/Voice/lay_emotes.yml]: no mention of "falls over"
i.e 

> Scotty Garland falls over!

File: Resources/Prototypes/_CD/Voice/lay_emotes.yml
Comment (DangerRevolution) [inline Resources/Prototypes/_CD/Voice/lay_emotes.yml]: probably could use "gets up"

File: Resources/Prototypes/_CD/Voice/lay_emotes.yml
Comment (Adrian16199) [reply Resources/Prototypes/_CD/Voice/lay_emotes.yml]: Look 23 lines below that

File: Resources/Prototypes/_CD/Voice/lay_emotes.yml
Comment (Adrian16199) [reply Resources/Prototypes/_CD/Voice/lay_emotes.yml]: Fair point

**Comment by LankLTE** [comment]
This system has some issues which I’ve mentioned on discord here. 
https://discord.com/channels/968983104247185448/1150555473699934370/1229875767043686410

There’s also a lack of other phrases as I hadn’t fully finished it as of you deciding to port this. As for an action, entirely possible (and pretty easy, actually), but I personally think that filling your sidebars with actions that all just do an emote is a bad idea. Especially when roles like engineers already frequently have 6-8 actions. You may disagree.

**Comment by Adrian16199** [comment]
Pretty much ready for review.

File: Resources/Prototypes/_CD/Voice/lay_emotes.yml
Comment (DEATHB4DEFEAT) [inline Resources/Prototypes/_CD/Voice/lay_emotes.yml]:
```suggestion
    - seizes up and falls limp, his eyes dead and lifeless...
    - seizes up and falls limp, her eyes dead and lifeless... # Delta V - Bit more options end here
```

File: Resources/Prototypes/_CD/Voice/lay_emotes.yml
Comment (DEATHB4DEFEAT) [inline Resources/Prototypes/_CD/Voice/lay_emotes.yml]:
```suggestion
    - lays
    - lays.
    - lays down
    - lays down.
    - lies down
    - lies down.
    - trips
    - trips.
    - trips!
    - slips
    - slips.
    - slips!
    - goes prone
    - goes prone.
    - crawls
    - crawls.
    - collapses
    - collapses.
    - collapses!
    - falls
    - falls.
    - falls!
    - falls down
    - falls down.
    - falls down!
    - falls over
    - falls over.
    - falls over!
    - gets pushed over
    - gets pushed over.
    - gets pushed over!
    - slips
    - slips.
    - slips!
    - collapses
    - collapses.
    - collapses!
```

File: Resources/Prototypes/_CD/Voice/lay_emotes.yml
Comment (DEATHB4DEFEAT) [inline Resources/Prototypes/_CD/Voice/lay_emotes.yml]:
```suggestion
    - stands
    - stands.
    - stands up
    - stands up.
    - gets up
    - gets up.
    - slowly gets up
    - slowly gets up.
    - slowly stands up
    - slowly stands up.
    - hops up
    - hops up.
```

File: Content.Server/_CD/Traits/LayEmoteSystem.cs
Comment (DEATHB4DEFEAT) [inline Content.Server/_CD/Traits/LayEmoteSystem.cs]: Make this a variable

```suggestion
            args.ModifySpeed(component.CrawlSpeed, component.CrawlSpeed); // Delta V- crawl, baby, crawl!
```

**Review by DEATHB4DEFEAT — CHANGES_REQUESTED** [review]

File: Content.Server/_CD/Traits/LayEmoteSystem.cs
Comment (Adrian16199) [reply Content.Server/_CD/Traits/LayEmoteSystem.cs]: Erm, wouldnt this change require me to make that component?

Comment (Adrian16199) [reply Content.Server/_CD/Traits/LayEmoteSystem.cs]: I guess I will just take it and start changin code just so *that* fits.

**Review by Axionyxx — APPROVED** [review]
I feel like I might be missing something, so I'd appreciate it if another maintainer also approved this prior to it being merged. I feel like some horrible bug will reveal itself after this is merged triggered by unique edge cases like the noospheric zap and just result in horrible problems, but I might be overthinking it

File: Content.Server/_CD/Traits/LayEmoteSystem.cs
Comment (deltanedas) [inline Content.Server/_CD/Traits/LayEmoteSystem.cs:14]: sort alphabetically + rename to not have System as its redundant

File: Content.Server/_CD/Traits/LayEmoteSystem.cs
Comment (deltanedas) [inline Content.Server/_CD/Traits/LayEmoteSystem.cs:38]: `if (!component.Laying) return;` then remove from the other if statements

File: Content.Server/_CD/Traits/LayEmoteSystem.cs
Comment (deltanedas) [inline Content.Server/_CD/Traits/LayEmoteSystem.cs:81]: else

File: Content.Server/_CD/Traits/LayEmoteSystem.cs
Comment (deltanedas) [inline Content.Server/_CD/Traits/LayEmoteSystem.cs:39]: this will still leave you laying down after unbuckling which is weird

File: Content.Server/_CD/Traits/LayEmoteSystem.cs
Comment (deltanedas) [inline Content.Server/_CD/Traits/LayEmoteSystem.cs:48]: i imagine this could possibly troll your movement speed, add a SetLaying function that 1. checks if arg isnt the same as Laying and 2. does the Stand Down update movespeed etc
then use that everywhere

then here call SetLaying(ent, false);

File: Content.Server/_CD/Traits/LayEmoteSystem.cs
Comment (deltanedas) [inline Content.Server/_CD/Traits/LayEmoteSystem.cs:29]: for all these handlers use `Entity<LayEmoteComponent>` and ref args

**Review by deltanedas — CHANGES_REQUESTED** [review]

**Comment by deltanedas** [comment]
test how this interacts with different kinds of stunning too

File: Content.Server/_CD/Traits/LayEmoteComponent.cs
Comment (Adrian16199) [reply Content.Server/_CD/Traits/LayEmoteComponent.cs:19]: what?

**Pull request closed without merging**
